src/wolfsftp.c

```diff
@@ -2899,38 +2858,24 @@ static int wolfSSH_SFTPNAME_readdir(WOLFSSH* ssh, WDIR* dir, WS_SFTPNAME* out,
 
     /* attempt to get file attributes. Could be directory or have none */
     {
-        char* buf;
-        int   bufSz;
-        int   tmpSz;
-
-        bufSz = out->fSz + (int)WSTRLEN(dirName) + sizeof(WS_DELIM);
-        buf = (char*)WMALLOC(bufSz + 1, out->heap, DYNTYPE_SFTP);
-        if (buf == NULL) {
-            WFREE(out->fName, out->heap, DYNTYPE_SFTP);
-            return WS_MEMORY_E;
-        }
-        buf[0] = '\0';
-        WSTRNCAT(buf, dirName, bufSz + 1);
-        tmpSz = (int)WSTRLEN(buf);
+        char  r[WOLFSSH_MAX_FILENAME];
+        char  s[WOLFSSH_MAX_FILENAME];
 
-        /* add delimiter between path and file/dir name */
-        if (tmpSz + 1 < bufSz) {
-            buf[tmpSz] = WS_DELIM;
-            buf[tmpSz+1] = '\0';
-        }
-        WSTRNCAT(buf, out->fName, bufSz + 1);
+        r[0] = '\0';
+        WSTRNCAT(r, dirName, sizeof(r));
```

Comment:
Same question as earlier about this pattern.

Reply:
Returning an error if everything won't fit into r, and freeing the out filename.